lumigator/python/mzai/backend/backend/api/routes/health.py

```diff
@@ -41,24 +42,63 @@ def get_job_metadata(job_id: UUID) -> JobSubmissionResponse:
     except json.JSONDecodeError as e:
         loguru.logger.error(f"JSON decode error: {e}")
         loguru.logger.error(f"Response text: {resp.text}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Invalid JSON response"
+        ) from e
+
+
+@router.get("/jobs/{job_id}/logs")
+def get_job_logs(job_id: UUID) -> JobLogsResponse:
+    resp = requests.get(urllib.parse.urljoin(settings.ray_jobs, f"{job_id}/logs"))
+
+    if resp.status_code == HTTPStatus.NOT_FOUND:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND,
+            detail=f"Job logs for ID: {job_id} not found - {resp.text or ''}",
+        )
+    elif resp.status_code == HTTPStatus.BAD_REQUEST:
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail=f"Job ID: {job_id} not submitted via the Ray API - {resp.text or ''}",
+        )
```

Comment:
If we're making a bad request to Ray, that's on 'us' the backend, not the caller, we should have verified we're happy with whatever parameters we're sending upstream beforehand. 

If the caller has supplied a UUID (which has to be valid otherwise Pydantic would have raised a validation error), why should they be told it was a bad request? It wasn't.

I think we should stick with letting other status codes fall through to the `500 - Internal Server Error` as we're doing something wrong not the caller.

Reply:
Thanks @peteski22 , I was not 100% sure about this but I thought it was covering some use case that Javier had encountered and I could not replicate. I agree that if it is something we do not foresee now (and it cannot be invalid UUID, which is the only parameter we are passing and is validated) we should just raise a 500.

Reply:
The UUID might be referring to a non-existing job, for example, or be a non-type 4 UUID. IMHO the final decision is on Ray and we should stick to their response. Also, we don't have an internal error. Our code did not fail or have an exception. Some other SW tells us this is a bad request, so IMHO we need to say that this was a bad request (and hopefully include useful info from Ray as to why this is so).

Reply:
As decided, 400 will translate to 500.